### PR TITLE
Add error checking in String::New function

### PR DESCRIPTION
**Pull request by JckXia**
Added an if statement to String::New(env, const char * str) to make sure that str pointer is not null

### CODE REVIEW

**Comment by JckXia** [comment]
Should I also add this if statement to the other` String::New `overloads that accept a C char pointer? Also  for the overload `String::New(env, const char *, length)`, it doesn't seem like there are any checks in place to make sure that the length param is indeed the length of the string passed in. If I pass a value that exceeds the length of the string ( in this case String::New(env, "abcde",46)), I get this result: 

stringLengthCheck
Is this expected behavior? Thank you! 
To reproduce the above result, I defined a C++ function returning String::New(env, "abcde",46). From the JS side, I printed out the return value of this Cpp function using `console.log()`

File: test/name.js [hunk 1]
```diff
@@ -6,7 +6,13 @@ module.exports = require('./common').runTest(test);
...
 function test(binding) {
   const expected = '123456789';
-
+  
+  try{
+    binding.name.nullStringShouldThrow();
+  }catch(e)
+  {
+    assert.strictEqual(e.message, 'String::New recieved a nullpointer as a value');
+  }
```
Comment (KevinEady) [inline test/name.js:15]: You can use `assert.throws()`

File: napi-inl.h [hunk 2]
```diff
@@ -900,6 +900,11 @@ inline String String::New(napi_env env, const std::u16string& val) {
 }
 
 inline String String::New(napi_env env, const char* val) {
+  if (val == nullptr) {
+    NAPI_THROW(
+        TypeError::New(env, "String::New recieved a nullpointer as a value"),
```
Comment (KevinEady) [inline napi-inl.h:905]: typo 😉

**Review by KevinEady — CHANGES_REQUESTED** [review]

**Review by KevinEady — APPROVED** [review]
LGTM

**Review by mhdawson — APPROVED** [review]
LGTM

**Comment by mhdawson** [comment]
Landed in da50b51398789185ab63ece534890f131703ffbb

**Pull request closed without merging**